pkg/landscaper/gardenlet/gardenlet.go

```diff
@@ -0,0 +1,170 @@
+
+	landscaper.gardenClient = gardenClient
+
+	// Create Seed client
+	seedClient, err := kubernetes.NewClientFromBytes([]byte(imports.RuntimeCluster.Spec.Configuration.Kubeconfig))
+	if err != nil {
+		return nil, fmt.Errorf("failed to create the runtime cluster client: %v", err)
+	}
+
+	landscaper.seedClient = seedClient
+
+	return landscaper, nil
+}
+
+func (g Landscaper) Run(ctx context.Context) error {
+	switch g.landscaperOperation {
+	case landscaperReconciliation:
+		return g.Reconcile(ctx)
+	case landscaperDeletion:
+		return g.Delete(ctx)
+	default:
+		return fmt.Errorf(fmt.Sprintf("environment variable \"OPERATION\" must either be set to %q or %q", landscaperReconciliation, landscaperDeletion))
+	}
+}
+
+// parseImageVectorOverride parses the image vectors as a string from the import values
+// the image vectors are already properly created by the landscaper and are only handed
+// over as-is to the Gardenlet helm chart
+func (g *Landscaper) parseImageVectorOverride() error {
+	if g.Imports.ImageVectorOverwrite != nil {
+		var imageVectorOverwrite string
+		if err := json.Unmarshal(g.Imports.ImageVectorOverwrite.Raw, &imageVectorOverwrite); err != nil {
+			return err
+		}
+		g.imageVectorOverride = &imageVectorOverwrite
+	}
+
+	if g.Imports.ComponentImageVectorOverwrites != nil {
+		var componentImageVectorOverwrites string
+		if err := json.Unmarshal(g.Imports.ComponentImageVectorOverwrites.Raw, &componentImageVectorOverwrites); err != nil {
+			return err
+		}
+		g.componentImageVectorOverwrites = &componentImageVectorOverwrites
+	}
+	return nil
+}
+
+// parseGardenletImage gets the Gardenlet image from the component descriptor
+// The component descriptor is the only image source and must be provided
+func (g *Landscaper) parseGardenletImage(componentDescriptorList *v2.ComponentDescriptorList) error {
+	// The function returns a list as there could be multiple components with the same name but different version
+	components := componentDescriptorList.GetComponentByName(gardenerComponentName)
+	if len(components) != 1 {
+		return fmt.Errorf(fmt.Sprintf("expecting exactly one component with name %q", gardenerComponentName))
+	}
+
+	// get gardenlet image from component descriptor
+	res := components[0].GetLocalResourcesByName(v2.OCIImageType, gardenletImageName)
+	if len(res) == 0 {
+		return fmt.Errorf("OCI image for gardenlet not found in component descriptor")
+	}
+	gardenletResource := res[0]
+	imageVersion := gardenletResource.GetVersion()
+	if len(imageVersion) == 0 {
+		return fmt.Errorf("OCI image version for gardenlet not found in component descriptor")
+	}
+	imageReference := gardenletResource.Access.(*v2.OCIRegistryAccess).ImageReference
+	if len(imageReference) == 0 {
+		return fmt.Errorf("OCI image reference for gardenlet not found in component descriptor")
+	}
+
+	// split version from reference e.g eu.gcr.io/gardener-project/gardener/gardenlet:v1.11.3
+	split := strings.Split(imageReference, ":")
+	if len(split) != 2 {
```

Comment:
we should allow more than 2 parts as a valid reference could be `myregistry.example.com:5000/gardener/gardenlet:v1.12.4`